python/cudaq/_cudaq.cpp

```diff
@@ -72,7 +72,18 @@ PYBIND11_MODULE(_pycudaq, mod) {
         cudaq::mpi::all_gather(global, local);
         return global;
       },
-      "Gather and scatter the `local` list, returning a concatenation of all "
+      "Gather and scatter the `local` list of floating-point numbers, "
+      "returning a concatenation of all "
+      "lists across all ranks. The total global list size must be provided.");
+  mpiSubmodule.def(
```

Comment:
What was the reason to have no binding for the mpi broadcast in C++? 
Python and C++ API should match (up to language specific naming conventions) when it comes to the CUDA Quantum API.

Reply:
It was a miss. I've added the Python binding.